ecs.tf

```diff
@@ -59,43 +51,70 @@ resource "aws_appautoscaling_target" "app_scale_target" {
   min_capacity       = var.ecs_autoscale_min_instances
 }
 
+module "app_container_definition" {
+  source                       = "git::https://github.com/cloudposse/terraform-aws-ecs-container-definition.git?ref=tags/0.21.0"
+  container_name               = var.container_name
+  container_image              = var.docker_image
+  container_cpu                = var.container_cpu
+  container_memory             = var.container_memory
+  essential                    = true
+  readonly_root_filesystem     = false
+  environment                  = var.environment_vars
+  port_mappings                = [
+    {
+      containerPort            = var.container_port
+      hostPort                 = var.container_port
+      protocol                 = "tcp"
+    },
+  ]
+  log_configuration            = {
+    logDriver = "awslogs"
+    options   = {
+      "awslogs-group" = "/fargate/service/${var.app}-${var.environment}"
+      "awslogs-region" = "eu-west-1"
+      "awslogs-stream-prefix" = "ecs"
+    }
+    secretOptions = null
+  }
+}
+
+module "datadog_container_definition" {
+  source                       = "git::https://github.com/cloudposse/terraform-aws-ecs-container-definition.git?ref=tags/0.21.0"
+  container_name               = "datadog-agent"
+  container_image              = "datadog/agent:latest"
+  container_cpu                = "10"
+  container_memory             = "128"
+  essential                    = true
+  readonly_root_filesystem     = false
+  environment                  = [
+   {
+      name = "DD_API_KEY",
```

Comment:
It might be nice to add `DD_TAGS` to set some global tags, or at least make it configurable. Any JVM apps using this for example will probably also want to add `SD_JMX_ENABLE` to collect JVM metrics.

Perhaps an optional parameter of Datadog settings would do the trick

Reply:
oh though also the JMX metrics need a different Datadog container image + a bunch of docker labels so maybe it is easier to cross that bridge if we ever do come to it

You might also need to expose 8125/udp and 8126 but I'm not sure if you need to do that to allow containers within the same task definition to communicate with each other? I've had mixed messages from the docs + experiences

Reply:
On the first one, agree lets cross the bridge. On the second note, I've tested the above code and it seems to work (sending CPU/Memory data). I did not see any docs saying to open those ports. I was following these instructions. https://docs.datadoghq.com/integrations/ecs_fargate/#aws-cli